packages/create-sitecore-jss/src/templates/angular-xmcloud/src/app/jss-link.service.ts

```diff
@@ -0,0 +1,25 @@
+import { Inject, Injectable } from '@angular/core';
+import { DOCUMENT } from '@angular/common';
+import { HTMLLink } from '@sitecore-jss/sitecore-jss-angular';
+
+@Injectable()
+export class JssLinkService {
+  document: Document;
+
+  constructor() {
+    this.document = Inject(DOCUMENT);
+  }
+
+  addHeadLinks(headLinks: HTMLLink[]) {
+    if (!headLinks || !headLinks.length) {
+      return;
+    }
+
+    headLinks.forEach((headLink: HTMLLink) => {
+      let link: HTMLLinkElement = this.document.createElement('link');
+      link.setAttribute('rel', headLink.rel);
+      link.setAttribute('href', headLink.href);
+      this.document.head.appendChild(link);
```

Comment:
Any concern about these accumulating when navigating to different routes client-side? Looking at comment here: https://stackoverflow.com/a/50725256